[Chassis] test_drop_counters failing on T2 chassis testbed with single and multi asic linecards

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/sonic-net/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

**Steps to reproduce the issue:**
1.  On a T2 testbed which has mix of single and multi asic linecards run the tests in file `test_drop_counters`
2. Majority of the test fail with following errors
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
drop_packets/test_drop_counters.py:248: in do_counters_test
    skip_counter_check=skip_counter_check, drop_information=drop_information)
drop_packets/test_drop_counters.py:106: in base_verification
    duthost.command(CMD_PREFIX+&quot;sonic-clear rifcounters&quot;)
common/devices/multi_asic.py:108: in _run_on_asics
    return getattr(self.sonichost, self.multi_asic_attr)(*module_args, **complex_args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = &lt;SonicHost str2-sonic-lc5-1&gt;
module_args = (&apos;sudo ip netns exec None sonic-clear rifcounters&apos;,)
complex_args = {}, previous_frame = &lt;frame object at 0x558f49b90280&gt;
filename = &apos;/azp/_work/29/s/tests/common/devices/multi_asic.py&apos;
line_number = 108, function_name = &apos;_run_on_asics&apos;
lines = [&apos;            return getattr(self.sonichost, self.multi_asic_attr)(*module_args, **complex_args)\n&apos;]
index = 0, verbose = True, module_ignore_errors = False, module_async = False

    def _run(self, *module_args, **complex_args):
    
        previous_frame = inspect.currentframe().f_back
        filename, line_number, function_name, lines, index = inspect.getframeinfo(previous_frame)
    
        verbose = complex_args.pop(&apos;verbose&apos;, True)
    
        if verbose:
            logging.debug(&quot;{}::{}#{}: [{}] AnsibleModule::{}, args={}, kwargs={}&quot;\
                .format(filename, function_name, line_number, self.hostname,
                        self.module_name, json.dumps(module_args), json.dumps(complex_args)))
        else:
            logging.debug(&quot;{}::{}#{}: [{}] AnsibleModule::{} executing...&quot;\
                .format(filename, function_name, line_number, self.hostname, self.module_name))
    
        module_ignore_errors = complex_args.pop(&apos;module_ignore_errors&apos;, False)
        module_async = complex_args.pop(&apos;module_async&apos;, False)
    
        if module_async:
            def run_module(module_args, complex_args):
                return self.module(*module_args, **complex_args)[self.hostname]
            pool = ThreadPool()
            result = pool.apply_async(run_module, (module_args, complex_args))
            return pool, result
    
        res = self.module(*module_args, **complex_args)[self.hostname]
    
        if verbose:
            logging.debug(&quot;{}::{}#{}: [{}] AnsibleModule::{} Result =&gt; {}&quot;\
                .format(filename, function_name, line_number, self.hostname, self.module_name, json.dumps(res)))
        else:
            logging.debug(&quot;{}::{}#{}: [{}] AnsibleModule::{} done, is_failed={}, rc={}&quot;\
                .format(filename, function_name, line_number, self.hostname, self.module_name, \
                        res.is_failed, res.get(&apos;rc&apos;, None)))
    
        if (res.is_failed or &apos;exception&apos; in res) and not module_ignore_errors:
&gt;           raise RunAnsibleModuleFail(&quot;run module {} failed&quot;.format(self.module_name), res)
E           RunAnsibleModuleFail: run module command failed, Ansible Results =&gt;
E           {&quot;changed&quot;: true, &quot;cmd&quot;: [&quot;sudo&quot;, &quot;ip&quot;, &quot;netns&quot;, &quot;exec&quot;, &quot;None&quot;, &quot;sonic-clear&quot;, &quot;rifcounters&quot;], &quot;delta&quot;: &quot;0:00:00.019185&quot;, &quot;end&quot;: &quot;2022-12-26 21:14:10.835705&quot;, &quot;failed&quot;: true, &quot;msg&quot;: &quot;non-zero return code&quot;, &quot;rc&quot;: 255, &quot;start&quot;: &quot;2022-12-26 21:14:10.816520&quot;, &quot;stderr&quot;: &quot;Cannot open network namespace \&quot;None\&quot;: No such file or directory&quot;, &quot;stderr_lines&quot;: [&quot;Cannot open network namespace \&quot;None\&quot;: No such file or directory&quot;], &quot;stdout&quot;: &quot;&quot;, &quot;stdout_lines&quot;: [], &quot;warnings&quot;: [&quot;Consider using &apos;become&apos;, &apos;become_method&apos;, and &apos;become_user&apos; rather than running sudo&quot;]}

complex_args = {}
filename   = &apos;/azp/_work/29/s/tests/common/devices/multi_asic.py&apos;
function_name = &apos;_run_on_asics&apos;
index      = 0
line_number = 108
lines      = [&apos;            return getattr(self.sonichost, self.multi_asic_attr)(*module_args, **complex_args)\n&apos;]
module_args = (&apos;sudo ip netns exec None sonic-clear rifcounters&apos;,)
module_async = False
module_ignore_errors = False
previous_frame = &lt;frame object at 0x558f49b90280&gt;
res        = {&apos;stderr_lines&apos;: [u&apos;Cannot open network namespace &quot;None&quot;: No such file or dire...nnot open network namespace &quot;None&quot;: No such file or directory&apos;, &apos;failed&apos;: True}
self       = &lt;SonicHost str2-sonic-lc5-1&gt;
verbose    = True
```
**Describe the results you received:**
 The testcase fail with above errors

**Describe the results you expected:**
The test case should run properly on T2 testbed with mix of single and multi asic linecards 

**Additional information you deem important:**
<!--
software/ASIC/Hardware Flatform version and info
-->
    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

--- a/tests/common/helpers/drop_counters/drop_counters.py
+++ b/tests/common/helpers/drop_counters/drop_counters.py
@@ -36,10 +36,10 @@ def get_pkt_drops(duthost, cli_cmd, asic_index):
     # Frame the correct cli command
     # the L2 commands need _SUFFIX and L3 commands need _PREFIX
     if cli_cmd == GET_L3_COUNTERS:
-        CMD_PREFIX = NAMESPACE_PREFIX if duthost.is_multi_asic else ''
+        CMD_PREFIX = NAMESPACE_PREFIX if (namespace is not None and duthost.is_multi_asic) else ''
         cli_cmd = CMD_PREFIX + cli_cmd
     elif cli_cmd == GET_L2_COUNTERS:
-        CMD_SUFFIX = NAMESPACE_SUFFIX if duthost.is_multi_asic else ''
+        CMD_SUFFIX = NAMESPACE_SUFFIX if (namespace is not None and duthost.is_multi_asic) else ''
         cli_cmd = cli_cmd + CMD_SUFFIX
 
     stdout = duthost.command(cli_cmd.format(namespace))

--- a/tests/drop_packets/test_drop_counters.py
+++ b/tests/drop_packets/test_drop_counters.py
@@ -60,8 +60,8 @@ def enable_counters(duthosts):
 
     yield
     for duthost in duthosts.frontend_nodes:
+        namespace_list = duthost.get_asic_namespace_list() if duthost.is_multi_asic else ['']
         for namespace in namespace_list:
-            namespace_list = duthost.get_asic_namespace_list() if duthost.is_multi_asic else ['']
             for port, status in previous_cnt_status[duthost][namespace].items():
                 if status == "disable":
                     logger.info("Restoring counter '{}' state to disable".format(port))
@@ -96,14 +96,14 @@ def base_verification(discard_group, pkt, ptfadapter, duthosts, asic_index, port
     Base test function for verification of L2 or L3 packet drops. Verification type depends on 'discard_group' value.
     Supported 'discard_group' values: 'L2', 'L3', 'ACL', 'NO_DROPS'
     """
-    # Clear SONiC counters
+    # Clear SONiC counters all the asic on all the duts
     for duthost in duthosts.frontend_nodes:
         duthost.command("sonic-clear counters")
-
-        # Clear RIF counters per namespace.
-        namespace = duthost.get_namespace_from_asic_id(asic_index)
-        CMD_PREFIX = NAMESPACE_PREFIX.format(namespace) if duthost.is_multi_asic else ''
-        duthost.command(CMD_PREFIX+"sonic-clear rifcounters")
+        namespace_list = duthost.get_asic_namespace_list() if duthost.is_multi_asic else ['']
+        for namespace in namespace_list:
+            # Clear RIF counters on all namespaces 
+            CMD_PREFIX = NAMESPACE_PREFIX.format(namespace) if duthost.is_multi_asic else ''
+            duthost.command(CMD_PREFIX+"sonic-clear rifcounters")
 
     send_packets(pkt, ptfadapter, ports_info["ptf_tx_port_id"], PKT_NUMBER)
 

